### ISSUE TITLE
Export for Flatpaks

### ISSUE THREAD

**Issue by Vovkiv** [vikdevelop/SaveDesktop#84]
(related to #40)
While i understand reasoning behind, i still want encourage you to implement exporting list of flatpak apps.
Apps (especially flatpak apps) is essential part of desktop, so storing list of them can be considered as part of DE.
Think of them as GNOME extensions, which is not really part of GNOME DE, but program saves them anyway.

SaveDesktop could export list of flatpaks like this `flatpak list --app --columns=origin` (if there flatpaks available to begin with)
And when importing from file done, it could copy to clipboard list as:
```
flatpak install -y a
flatpak install -y b
flatpak install -y c
flatpak install -y e
```
So user will be able to insert it into terminal and install.
Or, if there was backup of flatpak apps, it can show flatpak button which also copy this list to clipboard.
![image](https://github.com/vikdevelop/SaveDesktop/assets/54743395/2198138c-5f26-4bfa-9699-45afcdb1d402)

In any case, nice app.

**Comment by vikdevelop** [comment]
Hello, from what I've tried, with the org.freedesktop.Flatpak permission, it's not really a problem to list the applications and then write them to a file. The problem would be with installing flatpak apps from that list, it's a fairly long process, however when importing the configuration the user has to log out to successfully apply the changes anyway, so after logging back in the apps could be installed in the background.  So, I will try to implement it, but I am not sure, if it will be worked.

**Comment by Vovkiv** [comment]
Well, as i say, if this app will be able to export flatpaks list and then copy it to clipboard (so user can open terminal, past it and then press enter) at importing process, it will be already 100% useful.
Problem that i here see is not about how to install flatpaks, but rather how to get. You see, programs like this exist for easing process of making backups. And considering, that this app also can periodically make backups, it's make things perfect.

Instead of hacky shell scripts, i just click "do backup, please" and i'm done. But flatpaks also important thing. And i, as any user, sometimes install and uninstall them.
Right now if i want to store somewhere up-to-day list of installed flatpaks, i need to manually `flatpak list --app --columns=origin` to file, then with some shell scripting i need to  apply "flatpak install -y" to start for every item in list (so i can simply run that script). And THEN save file somewhere.
And i need to do this manually, instead of program that specifically designed to do such things!

In short, maintaining and exporting of flatpaks list is problem, not installing them, since you need to save them regularly while install maybe once a year when you, as user, *accidentally* broke system so you install fresh one and import setting and apps list.

**Comment by Vovkiv** [comment]
Also, this approach, not directly, but fixes issue that to import settings, you need to install SaveDesktop app. Since backup is stored as tar file, you will be able to extract it (almost any DE and linux distro include zip archiver for that out-of-box), run .sh file with all flatpaks (where SaveDesktop also will be) to install them and then run this app and import other DE options.

**Comment by vikdevelop** [comment]
I implemented option of saving installed Flatpak apps. Just click on switch "Save installed Flatpak apps" and click on Save button. After saving configuration is available bash script `installed_flatpaks.sh` in the Tar archive. This option for now is not available in the stable build, but is available in the test build. If you want try it, use this commands:
```bash
git clone https://github.com/vikdevelop/SaveDesktop
cd SaveDesktop
flatpak-builder build *.yaml --install --user
```
In terms of periodic saving, SaveDesktop creates same bash script automatically. When importing configuration, SaveDesktop moves script to the app's data directory and after the next login, it runs on background and installs Flatpak apps. Of course, it is available option unzip Tar archive and run the bash script.

**Comment by Vovkiv** [comment]
I tested it and i already like it, thanks!

**Comment by vikdevelop** [comment]
I uploaded on Flathub new version that has option for saving installed Flatpak apps. Sadly, with permission org.freedesktop.Flatpak flathub builder generates errors, so I decided use permission` filesystem=/var/lib/flatpak:ro` and edit some trifles. With this permission, SaveDesktop can create only list of apps, that installed to system directory (not home directory).
